backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java

```diff
@@ -211,110 +211,128 @@ private ExperimentRun getExperimentRunFromRequest(
-                            newExperimentRun.getMetricsList());
-                      }
-                      if (!newExperimentRun.getObservationsList().isEmpty()) {
-                        observationHandler.logObservations(
-                            handleForTransaction,
-                            newExperimentRun.getId(),
-                            newExperimentRun.getObservationsList(),
-                            now);
-                      }
-                      if (!newExperimentRun.getArtifactsList().isEmpty()) {
-                        artifactHandler.logArtifacts(
-                            handleForTransaction,
-                            newExperimentRun.getId(),
-                            newExperimentRun.getArtifactsList(),
-                            false);
-                      }
-                      if (!newExperimentRun.getFeaturesList().isEmpty()) {
-                        featureHandler.logFeatures(
-                            handleForTransaction,
-                            newExperimentRun.getId(),
-                            newExperimentRun.getFeaturesList());
-                      }
-                      if (newExperimentRun.getCodeVersionSnapshot().hasCodeArchive()
-                          || newExperimentRun.getCodeVersionSnapshot().hasGitSnapshot()) {
-                        codeVersionHandler.logCodeVersion(
-                            handleForTransaction,
-                            LogExperimentRunCodeVersion.newBuilder()
-                                .setId(newExperimentRun.getId())
-                                .setCodeVersion(newExperimentRun.getCodeVersionSnapshot())
-                                .setOverwrite(false)
-                                .build());
-                      }
-                      if (!newExperimentRun.getDatasetsList().isEmpty()) {
-                        datasetHandler.logArtifacts(
-                            handleForTransaction,
-                            newExperimentRun.getId(),
-                            newExperimentRun.getDatasetsList(),
-                            false);
-                      }
-                      if (newExperimentRun.getVersionedInputs().getRepositoryId() != 0) {
-                        versionInputHandler.validateAndInsertVersionedInputs(
-                            handleForTransaction,
-                            newExperimentRun.getId(),
-                            newExperimentRun.getVersionedInputs());
-                      }
-                    }))
+    return InternalFuture.completedInternalFuture(true)
+        .thenCompose(
+            unused -> {
+              if (newExperimentRun.getVersionedInputs().getRepositoryId() != 0) {
+                return versionInputHandler.validateVersioningEntity(
+                    newExperimentRun.getVersionedInputs());
+              }
+              return InternalFuture.completedInternalFuture(new HashMap<>());
```

Comment:
So this is the core of the change, right?

Reply:
Yes